[self-guided] tour pages

dvdherron opened the pull request:
## Cute animal pic
![image](https://user-images.githubusercontent.com/41588129/117682713-a460c600-b1b3-11eb-9ca7-9a6eaae1462e.png)

_Because everyone likes pictures of [animals](https://unsplash.com/t/animals)._



## Link to Trello card (if applicable)
https://trello.com/c/8oJo98Ex/261-as-a-new-user-i-want-to-view-more-information-about-the-metecho-ui-in-a-nonlinear-fashion-self-guided-tour-on-non-project-detail
_If this PR relates to a Trello card, don't forget to reference this PR on the card as well._



## Description
_The commit messages say what you did; this should explain why and/or how._
- [x ] Description is in Trello card



## Steps to test/reproduce

- Turn on the self-guided tour.
- Look for and open the tour popovers on the following pages
  - Project List
  - Epic Detail
  - Task Detail
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._



## Show me
### Project List
![image](https://user-images.githubusercontent.com/41588129/118318472-ba46f180-b4f9-11eb-9582-502b104246c4.png)

### Epic Detail
![image](https://user-images.githubusercontent.com/41588129/118318573-dd71a100-b4f9-11eb-958b-5fbc40116a2f.png)

### Task Detail
![image](https://user-images.githubusercontent.com/41588129/118318633-f37f6180-b4f9-11eb-9d61-65ad2c65d932.png)

_Provide screenshots/animated gifs/videos if necessary._

dvdherron commented:
@jgerigmeyer These are mostly in place. I think it can be ready for a review soon?

There are two that are missing: one for the Commit History and the one for the breadcrumb if a Task is not grouped in an Epic. 

- How do I get the commit history component to render?
- For the breadcrumb, do you see a way to better way to place the popovers there in order to have them appear over specific items in that list?

There's sort of a similar placement issue on the project list page. I have it on the left side of the entire list for now unless there's a way to put it only on the first item in that whole list.

jgerigmeyer commented on src/js/components/epics/detail.tsx:
```diff
@@ -372,19 +374,47 @@ const EpicDetail = (props: RouteComponentProps) => {
+          >
+            {branchLinkText}
+          </ExternalLink>
+          <TourPopover
+            align="bottom right"
+            heading={i18n.t('View GitHub branch for Epic')}
+            body={
+              <Trans i18nKey="tourViewEpicBranch">
+                Select this button to leave Metecho and access the Epic’s branch
+                on GitHub. A “branch” in Git is a way to create a new feature or
+                make a modification to existing software but not affect the main
+                “trunk” of the Project. A branch is created in GitHub when a new
+                Epic or Task is created in Metecho.
+              </Trans>
+            }
+          />
```
Both here (epic detail) and on task detail, we want this popover to change depending whether the button is "View Branch", "View Changes", or "View Pull Request". You can see the logic for this in `helpers.ts`, `getBranchLink` fn. Maybe we want a similar fn to get the right popover? Or we could repurpose `getBranchLink` to also return the popover?

dvdherron replied:
A little stuck here. Having a hard time figuring out the syntax to test whether the object is an epic or a task. Is what I'm doing here so far even close to being on the right track?

jgerigmeyer replied:
@dvdherron Yeah, I agree this is a bit tricky, but what you have is definitely on the right track. There would be a number of different ways to approach this, but I'd say the clearest might be adding a second argument that `getBranchLink` accepts: `type: 'epic' | 'task'`. Then I think in the fn body, you could do e.g. `taskPopoverHeading = type === OBJECT_TYPES.EPIC ? i18n.t('View GitHub Branch for Epic') : i18n.t('View GitHub Branch for Task');` -- or something along those lines?

Happy to pair with you if that helps. I'd also say if you get the logic working but are struggling to make TypeScript warnings happy, don't spend much time on that. Just commit and I can look closer at the TS.

dvdherron replied:
Thanks. My meeting went longer than planned. Digging into this now.

dvdherron replied:
was able to get this to work. will give it another look in the AM to see if there's a better way @jgerigmeyer

jgerigmeyer reviewed (commented):
@dvdherron This is looking good! I didn't do a comprehensive code review, but I did functional review and tried to answer your questions.

- For the commit history, if there are commits on the relevant GH branch (you can make these manually in GH outside Metecho if that's easier), you can run `derrick manage resync_all_gh_commit_data` or (`python manage.py resync_all_gh_commit_data` inside a container terminal) to get the list of commits to show up in Metecho.
- For the breadcrumb, skip the "no epic" task breadcrumb for now. That's not something we currently support.

If any of my comments don't make sense, I'm happy to jump on a call and screenshare in more detail!

dvdherron commented:
> 
> 
> @dvdherron This is looking good! I didn't do a comprehensive code review, but I did functional review and tried to answer your questions.
> 
>     * For the commit history, if there are commits on the relevant GH branch (you can make these manually in GH outside Metecho if that's easier), you can run `derrick manage resync_all_gh_commit_data` or (`python manage.py resync_all_gh_commit_data` inside a container terminal) to get the list of commits to show up in Metecho.
> 
>     * For the breadcrumb, skip the "no epic" task breadcrumb for now. That's not something we currently support.
> 
> 
> If any of my comments don't make sense, I'm happy to jump on a call and screenshare in more detail!

Fantastic. Thanks for the feedback! Working through these issues now and I'll circle back if I run into any trouble.

dvdherron commented on src/js/components/tour/popover.tsx:
```diff
@@ -12,7 +12,7 @@ const TourPopover = ({
 }: {
   align: string;
-  heading: string;
+  heading: string | undefined;
```
@jgerigmeyer There are no popovers without headings but without this the popover in `orgActions` causes an error. I'm [changing the header](https://github.com/oddbird/Metecho/pull/1088/files#diff-961f0674b32dc07637039a492502239afd654e6619acffc746c551632ba92f75R173) within the switch statement. Is there a better way to do that?

jgerigmeyer replied:
Ah, yeah -- I see, that makes sense. I think we can remove this, and I'll comment on that other section of code with a suggestion.

dvdherron commented on src/js/components/utils/detailPageLayout.tsx:
```diff
@@ -66,34 +68,49 @@ const DetailPageLayout = ({
+                Projects.
+              </Trans>
+            }
+          />
```
Adjusted this in a way that adds the respective popover for the epic detail and task detail pages. had to remove the "slds-truncate" class to get the popover to show. so far I don't see that affecting anything but if you know why that might be an issue I can rethink this.

jgerigmeyer replied:
Hm, I'd want to involve @stacyk or @SondraE in that discussion. It'll mean that longer names will wrap on narrower screens, which also causes some issues with the popover placement.

![Screen Shot 2021-05-18 at 12 14 06 PM](https://user-images.githubusercontent.com/552316/118687118-a853a000-b7d2-11eb-8226-12bd47203bd4.png)

stacyk replied:
@dvdherron @jgerigmeyer is it possible to wrap only the last child in the breadcrumb and place the popover icon at the beginning of the breadcrumb label it is documenting? Or maybe no need to wrap, but still keep the popover before?

dvdherron replied:
@stacyk Yes, that's possible. Something like this @stacyk 

![image](https://user-images.githubusercontent.com/41588129/118969889-e76a2880-b96d-11eb-892c-34deee90325c.png)

 As of right now, that would just require a change in positioning in the CSS. 

The tricky part is making sure the popover is visible with the ` slds-truncate` class added back to the div wrapping the `Breadcrumb` component. The `overflow: hidden` setting cuts off the popover contents:

![image](https://user-images.githubusercontent.com/41588129/118971299-8a6f7200-b96f-11eb-9564-87e3d7a3372f.png)

jgerigmeyer commented on src/js/components/orgs/cards/orgActions.tsx:
```diff
@@ -138,23 +140,40 @@ const OrgActions = ({
     switch (type) {
       case ORG_TYPES.DEV:
         isActive = hasReviewRejected || !task.has_unmerged_commits;
+        popoverHeading = i18n.t('Create a Dev Org');
+        popoverKey = 'tourTaskCreateDevOrg';
+        popoverBody =
+          'A Dev Org is a temporary Salesforce org where you can make changes that you would like to contribute to the Project. To create an Org, make sure you are connected to your Salesforce account with a Dev Hub enabled. Use the drop down menu to delete the Org when you no longer need it.';
```
It might be cleaner to define `let popover = null;` above, then do something like:

```suggestion
        popover = (
          <TourPopover
            align="right"
            heading={i18n.t('Create a Dev Org')}
            body={(
              <Trans i18nKey="tourTaskCreateDevOrg">
                A Dev Org is a temporary Salesforce org where you can make changes that you would like to contribute to the Project. To create an Org, make sure you are connected to your Salesforce account with a Dev Hub enabled. Use the drop down menu to delete the Org when you no longer need it.
              </Trans>
            )}
          />
        );
```

Then you can define `popover` differently for a QA org, and use it below with just `{popover}` after the button.

jgerigmeyer commented on src/js/components/tour/popover.tsx:
```diff
@@ -12,7 +12,7 @@ const TourPopover = ({
 }: {
   align: string;
-  heading: string | undefined;
+  heading?: string;
```
```suggestion
  heading: string;
```

jgerigmeyer commented on src/js/components/utils/detailPageLayout.tsx:
```diff
@@ -89,6 +88,14 @@ const DetailPageLayout = ({
                     </Link>
                   );
                 }
+                if (crumb.tourPopover) {
+                  return (
+                    <div className="slds-p-horizontal_x-small" key={idx}>
+                      {crumb.name}
+                      {crumb.tourPopover}
+                    </div>
+                  );
+                }
```
I think it would work to just always include `{crumb.tourPopover}` and remove this conditional; if it's `undefined` then it won't render anything anyway.

Also, not a big deal but for consistency I'd say we might as well include `{crumb.tourPopover}` in the above `<Link>` version too.

dvdherron commented on src/js/components/epics/detail.tsx:
```diff
@@ -372,19 +374,47 @@ const EpicDetail = (props: RouteComponentProps) => {
-          modelType={OBJECT_TYPES.EPIC}
-          handleOptionSelect={handlePageOptionSelect}
-        />
+        <div className="slds-is-relative inline-container vertical-right">
```
added
![image](https://user-images.githubusercontent.com/41588129/118801425-4a41be00-b8a1-11eb-8f48-d4c2fd040bb6.png)

dvdherron commented on src/js/components/tasks/detail.tsx:
```diff
@@ -621,6 +651,20 @@ const TaskDetail = (props: RouteComponentProps) => {
     >
       <DetailPageLayout
         title={task.name}
+        titlePopover={
```
Added these: 

capture: 
![image](https://user-images.githubusercontent.com/41588129/118801713-9987ee80-b8a1-11eb-9686-6589a4851c8d.png)

submitButton
![image](https://user-images.githubusercontent.com/41588129/118801948-dbb13000-b8a1-11eb-92e3-6701ebdfc54a.png)

dvdherron commented on src/js/components/projects/list.tsx:
```diff
@@ -95,7 +96,19 @@ const ProjectList = withScroll(({ y }: ScrollProps) => {
+          />
           {projects.map((project) => (
             <ProjectListItem project={project} key={project.id} />
           ))}
```
That worked 👍🏽 
![image](https://user-images.githubusercontent.com/41588129/118810767-8b8b9b00-b8ac-11eb-832b-d0759c28e7c7.png)

jgerigmeyer commented on src/js/components/orgs/cards/orgActions.tsx:
```diff
@@ -138,23 +140,76 @@ const OrgActions = ({
     return (
       <>
-        {submitReviewBtn}
+        {submitReviewBtn ? (
+          <span className="slds-is-relative">
+            {submitReviewBtn}
+            <TourPopover
+              align="right"
+              heading={i18n.t('Submit a review')}
+              body={
+                <Trans i18nKey="tourTaskSubmitReview">
+                  When you’re finished viewing and testing all the changes, come
+                  back to Metecho to leave your review. You will have the option
+                  to approve the work or request changes. Clearly describe any
+                  further changes you’d like the Developer to make.
+                </Trans>
+              }
+            />
+          </span>
+        ) : null}
```
This is pretty minor, but it looks like when both of these buttons are visible, the "Create Org" btn appears on top of the "Submit Review" popover icon:

![Screen Shot 2021-05-24 at 12 05 13 PM](https://user-images.githubusercontent.com/552316/119375323-ad0fcc80-bc88-11eb-97fc-41272d9b35e7.png)

dvdherron replied:
I'll work on adding this space when these appear together.

jgerigmeyer replied:
Ok -- if it's not simple then we can ignore it; it's hardly noticeable.

dvdherron replied:
Fixing the z-index helped here. re-positioned the popovers a bit too. 
![image](https://user-images.githubusercontent.com/41588129/119496704-0e589e00-bd64-11eb-8cdf-701aa44885e6.png)

jgerigmeyer commented on src/js/components/utils/detailPageLayout.tsx:
```diff
@@ -66,34 +67,38 @@ const DetailPageLayout = ({
-            breadcrumb.map((crumb, idx) => {
-              if (crumb.url) {
+      <div>
+        <div
+          className="slds-p-horizontal_x-large
+            slds-p-top_x-small
+            metecho-breadcrumb
+            slds-is-relative"
+        >
```
Just a reminder that this breadcrumb styling is not finalized. I know mobile is a low priority for Metecho, but things get pretty weird on narrow screens...

![Screen Shot 2021-05-24 at 12 06 23 PM](https://user-images.githubusercontent.com/552316/119375573-f829df80-bc88-11eb-8ae8-9f69018633e5.png)

dvdherron replied:
And revisit this

dvdherron replied:
What we discussed worked for getting the breadcrumb popover to show and to apply the `slds-truncate `styles to the navigation list items. 
![image](https://user-images.githubusercontent.com/41588129/119551366-dbc99800-bd99-11eb-860e-ddc4c16c4ea5.png)

@stacyk @jgerigmeyer 
I think this is looking better. The list items aren't wrapping and the popover for the page title moves to the left on narrow screens.

jgerigmeyer reviewed (commented):
@dvdherron This is looking good! I'll work on backfilling JS tests.

dvdherron commented on src/js/components/projects/list.tsx:
```diff
@@ -151,14 +187,29 @@ const ProjectList = withScroll(({ y }: ScrollProps) => {
+                        updated list of Projects.
+                      </Trans>
+                    }
+                  />
```
adjusted this. 
![image](https://user-images.githubusercontent.com/41588129/119392962-ac028d80-bcd0-11eb-8da9-9123914a3548.png)

And I think these two were placed in a similar way. Adjusted these also. 
![image](https://user-images.githubusercontent.com/41588129/119393098-ca688900-bcd0-11eb-83cc-09cc0a0f770b.png)

stacyk commented on src/js/components/epics/progress.tsx:
```diff
@@ -25,6 +27,16 @@ const EpicProgress = ({ range }: Props) => {
         thickness="small"
         color={value === 100 ? 'success' : undefined}
       />
+      <TourPopover
+        align="right"
+        heading={i18n.t('Task progress bar')}
+        body={
+          <Trans i18nKey="tourTaskProgress">
+            A quick reference to show what percentage of an Epic is complete
+            based on the number of Tasks that are complete.
+          </Trans>
+        }
+      />
     </div>
```
I don't know the best place to leave this @dvdherron, but I see tis progress bar description is also showing up where the progress path description should be. 
![task-progress-path](https://user-images.githubusercontent.com/1581694/119443322-0dd1f000-bcef-11eb-9337-3366343b1f86.jpg)

dvdherron replied:
Good catch. Both elements had the same translation key. Fixed. 
![image](https://user-images.githubusercontent.com/41588129/119477574-5c16db80-bd4f-11eb-81e4-4db80eccac7e.png)

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -37,12 +37,40 @@
 .popover-wrapper {
   position: absolute;
-  top: -10px;
+  top: -12px;
```
![zindex](https://user-images.githubusercontent.com/1581694/119443556-6b663c80-bcef-11eb-925f-34b16b06e781.jpg) 

If we bump this z-index to at least 1, it won't lay behind this table cell neighbor I believe

dvdherron replied:
fixed

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -37,12 +37,40 @@
     top: unset;
   }
 
+  .metecho-breadcrumb & {
+    left: 0;
+    top: -15px;
+  }
+
+  .next-steps & {
+    left: 200px;
+  }
+
+  .project-placeholder & {
+    top: 15px;
+  }
+
+  .right-edge & {
+    right: 0;
+    top: -8px;
+  }
+
+  .vertical-right & {
+    right: -6px;
+    top: -20px;
+  }
+
   .page-title & {
     top: 25px;
```
@dvdherron There are quite a few "magic numbers" in this file and I wonder if there is a more systematic way of positioning these? Maybe not, but it may be helpful to see some custom properties or variables https://www.lightningdesignsystem.com/design-tokens/#category-spacing

dvdherron replied:
@stacyk Made an attempt to clean these up with tokens. Since they are all being contained in different ways, it was tough to come up with a system that would work across the board. Let me know if this helps or not.

stacyk reviewed (commented):
@dvdherron I didn't quite make it through the breadcrumbs, but we can chat after standup. I left a few comments in here, this is looking super good. I noticed some of the text is different from the Self Guided Text file. I don't know if that is a problem or not, but I thought I would mention it.

jgerigmeyer commented:
> I noticed some of the text is different from the Self Guided Text file. I don't know if that is a problem or not, but I thought I would mention it.

@stacyk I did change some of the text intentionally during my review. Mostly three types:

- Where I found the specific wording to be confusing
- Correcting inaccuracies
- Grammatical consistency (e.g. `and` vs `&` in body text vs headings)

If there are specific copy changes that anyone questions, I'm happy to revisit!

jgerigmeyer commented on src/sass/components/_tour.scss:
```diff
@@ -55,6 +92,7 @@
   box-shadow: 0 0 2px $palette-violet-65;
   color: $palette-violet-65;
   transform: scale(1); // needed to prevent subpixel rounding in Safari
+  z-index: 1;
```
Similar to @stacyk's other comment, but I think we could use `$z-index-default` here: https://www.lightningdesignsystem.com/design-tokens/#category-z-index

dvdherron replied:
fixed

dvdherron commented on src/js/components/utils/detailPageLayout.tsx:
```diff
@@ -35,8 +38,40 @@ const DetailPageLayout = ({
+  const heading = i18n.t('Navigation breadcrumb');
+
+  popover =
+    type === OBJECT_TYPES.EPIC ? (
```
@jgerigmeyer Moved the breadcrumb popover here. Why isn't the text changing? Am I referencing `OBJECT_TYPES` incorrectly?

jgerigmeyer replied:
You were _accepting_ a `type` prop here, but not passing one in from where it's used. Fixed in https://github.com/oddbird/Metecho/commit/56cfb220d46b81aade5d2123aebe659e9a51d7da

dvdherron replied:
Aha. Got it. Thanks!

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -37,24 +37,73 @@
+    top: -($var-spacing-x-small);
+  }
+
+  .next-steps & {
```
@jgerigmeyer This app is being translated into other languages, right?  If so, then we won't be able to position these popovers based on where the current English text ends up, like this one. 

@dvdherron I think for block level elements like headings, we probably want to position the popover target before the content.

jgerigmeyer replied:
Yes, I think that's correct. We shouldn't do any positioning strictly based on the length of text (in current font and language).

dvdherron replied:
Makes sense. For consistency's sake then we should probably have all of the popovers placed on the same general side of the elements they are referencing, right? @stacyk 

I'm working on removing them from inside of heading elements based on your comment below.

stacyk commented on src/js/components/commits/list.tsx:
```diff
@@ -100,9 +102,22 @@ const CommitList = ({ commits }: { commits: Commit[] }) =>
+          slds-is-relative"
       >
         Commit History
+        <TourPopover
```
@mirisuzanne looking for a second opinion here, but I think section heading elements only accept phrasing content, not divs. Is that correct? 

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements

![image](https://user-images.githubusercontent.com/1581694/119557585-fd109100-bd65-11eb-9985-767ba4c29c90.png)

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -37,24 +37,73 @@
   }
 }
 
+.metecho-breadcrumb ol {
```
Do you know @jgerigmeyer if we ever did truncate each breadcrumb? I looked at staging and we don't do it there. 

@dvdherron One option (new story I assume) I see SLDS has an overflow option https://www.lightningdesignsystem.com/components/breadcrumbs/#With-Overflow 

Another idea **If we want to truncate each `<li>` separately**, which I think is better usability, we would need to do something like 
- updating this line to `.metecho-breadcrumb li` instead of `ol`
- overriding the `<a>` to not have `display: block` 
- optional:  `.metecho-breadcrumb li:not(:first-of-type)` so it doesn't truncate home.
- we'd need to add a way to target the last item, which isn't a link, to tell that div in that last li not to display block... `.metecho-breadcrumb .slds-breadcrumb__item:last-of-type > div { display: inline (or contents) }` ... that is unless we can change it to a span?

jgerigmeyer replied:
> Do you know @jgerigmeyer if we ever did truncate each breadcrumb? I looked at staging and we don't do it there.

I'm not sure -- I'm guessing we didn't?

dvdherron replied:
I tried truncating the `<li>`'s separately yesterday but having `overflow-hidden` cut of parts of the list markers and made them all really small @stacyk. I'll play around with this some more in a bit after I reposition the popovers

dvdherron commented:
@jgerigmeyer @stacyk 
Relocated these to the left-hand side of the page. The task progress bar popover is the last one that needs to be placed. 

![image](https://user-images.githubusercontent.com/41588129/119705295-db440680-be58-11eb-90a2-bc4d90976bf7.png)

I'll get to this in the AM.

dvdherron commented:
@jgerigmeyer I think it's a Python test coverage that's failing? I see 100% on all of the JS tests locally. 

@stacyk we addressed the breadcrumb issue yesterday. I think I addressed the rest of the issues above also if you want to give this another review.

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -37,14 +37,73 @@
+    top: $var-spacing-medium;
+  }
+
+  .user-info & {
+    left: -($spacing-xx-small);
+    top: -($spacing-small);
+  }
```
@dvdherron this can likely be removed if the suggestion for logo-popover works.

dvdherron replied:
removed

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -37,14 +37,73 @@
+  }
 
   .logo-popover & {
-    top: unset;
+    top: $spacing-small;
+    left: $spacing-large;
+  }
```
@dvdherron If we target `.logo-popover > .popover-wrapper` then we don't need to also include the override for `.user-info &`

Also, these 2 properties can be re-ordered by alpha.

dvdherron replied:
done. didn't need that override there.

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -37,14 +37,73 @@
+    top: -($spacing-small);
+  }
+
+  .page-options & {
+    left: -($spacing-xx-small);
+    top: -($spacing-medium);
+  }
```
@dvdherron same here, we could target `.page-title > .popover-wrapper` and then we probably won't need this. Although, I notice we are using a `-1rem` top value as opposed the the `.75rem` that other buttons use. Is there a reason for the `.25` difference?

dvdherron replied:
@stacyk I had originally thought it looked better to get all three of these on a similar level but I think the slight difference on narrow screens is actually fine. 

![image](https://user-images.githubusercontent.com/41588129/119972451-861c0800-bfb2-11eb-9173-45f516a44f57.png)

stacyk replied:
me too @dvdherron

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -37,14 +37,73 @@
   }
 
   .page-title & {
-    top: 25px;
+    left: $spacing-large;
+    top: $var-spacing-medium;
+  }
```
@dvdherron we can try `.page-title > .popover-wrapper` to only target the direct children of the page-title container

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -37,14 +37,73 @@
+
+  .heading & {
+    top: -($spacing-medium);
+    left: -($spacing-xx-small);
```
@dvdherron I think this is default, and not needed here? If it is needed, then we will want to reorder properties but I think we can remove the left: ...

dvdherron replied:
yes. the `left` property was redundant. removed.

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -37,14 +37,73 @@
+    top: -($spacing-medium);
+  }
+
+  .vertical-right & {
+    right: -($spacing-x-small);
+    top: -($spacing-medium);
+  }
```
@dvdherron This is another button that has a slightly different positioning from some of the other buttons. If there is a reason to keep this one different than the others, maybe we can document why there are some that benefit from the position overrides. If this isn't needed, we will want to remove it along with the place it is used in the list.tsx file since I think that is the only place it is used.

dvdherron replied:
this was something I missed from the transition from right to left placement. removed.

stacyk commented on src/js/components/epics/detail.tsx:
```diff
@@ -486,16 +554,62 @@ const EpicDetail = (props: RouteComponentProps) => {
         }
       >
-        <EpicStatusPath status={epic.status} prIsOpen={epic.pr_is_open} />
+        <div className="slds-is-relative right-edge">
```
@dvdherron are we using the right-edge class for anything?

dvdherron replied:
same as above. removed.

stacyk commented on src/js/components/orgs/taskOrgCard.tsx:
```diff
@@ -175,12 +177,53 @@ const TaskOrgCard = ({
         slds-large-size_1-of-2
-        slds-p-around_x-small"
+        slds-p-around_x-small
+        slds-is-relative right-edge"
```
@dvdherron if we aren't using right-edge anywhere (currently on the task and epic status sidebars I think) then we could use something on the task cards to push them a little closer to their corner's edge for consistency? Although, we may want a different classname than `right-edge` since that doesn't seem to indicate what it is doing.

dvdherron replied:
made an `.org-card` class and moved them closer

![image](https://user-images.githubusercontent.com/41588129/119975666-6c7cbf80-bfb6-11eb-8242-338f715329a6.png)

stacyk commented on src/js/components/orgs/cards/orgActions.tsx:
```diff
@@ -138,23 +140,76 @@ const OrgActions = ({
-            variant={isActive ? 'brand' : 'neutral'}
-            onClick={doCreateOrg}
-          />
+          <span className="slds-is-relative task-create-btn">
```
@dvdherron if we add `inline-container` here, then do we need the popover override for `task-create-btn`?

dvdherron replied:
adding `inline-container` worked. removed `task-create-btn`

stacyk reviewed (commented):
@dvdherron this is looking really good. I left some suggestions about some ways that I think could make things more consistent/simplified but you will want to test them out on your end to see if they make sense in context. I didn't have all the scenarios so some of my suggestions may not take certain states into account.

jgerigmeyer commented on src/js/components/commits/list.tsx:
```diff
@@ -97,13 +99,27 @@ TimestampTableCell.displayName = DataTableCell.displayName;
-      >
-        Commit History
-      </h2>
+      <div className="slds-is-relative heading">
```
@dvdherron Maybe this is over-engineering, but I wonder if it would be cleaner for the `<TourPopover>` component itself to render its own `<div className="slds-is-relative">` container, and to also accept an optional `className` prop to add e.g. `className="heading"` or whatever. It would sure make the diff simpler from the places where we're adding in popovers... but I don't know if a single wrapping `div` would work in every case?

jgerigmeyer replied:
Oh, I guess that wouldn't really work because the popover _and_ some arbitrary contents need to be wrapped together... Probably simpler to just leave it as-is then 👍

dvdherron replied:
yeah, inserting it to each respective reference element probably has to be done manually with this type of popover . . .

jgerigmeyer reviewed (commented):
@dvdherron This is looking really good!

dvdherron commented on src/js/components/epics/progress.tsx:
```diff
@@ -12,12 +14,22 @@ const EpicProgress = ({ range }: Props) => {
   const value = getPercentage(complete, total);
   return (
-    <div className="epic-process">
+    <div className="epic-progress slds-is-relative">
```
yes, this is extra. removed.

jgerigmeyer commented on src/js/components/orgs/taskOrgCard.tsx:
```diff
@@ -179,8 +208,10 @@ const TaskOrgCard = ({
         slds-large-size_1-of-2
-        slds-p-around_x-small"
+        slds-p-around_x-small
+        slds-is-relative org-card"
```
```suggestion
        slds-is-relative
        org-card"
```

stacyk commented on src/sass/components/_tour.scss:
```diff
@@ -75,3 +115,7 @@
   text-align: left;
   white-space: initial;
 }
+
+.inline-container {
+  display: inline-block;
+}
```
@dvdherron I'm not sure what happened to my comment on this one so forgive me if it is duplicate, but I think this could move to one of the display (data-helpers or containers) scss files in the patterns folder. It feels like a more generic pattern that can be used elsewhere even though we are currently only using it on tour items at this time.

dvdherron replied:
👍🏽  moved to `_containers`

stacyk reviewed (approved):
@dvdherron One more tiny change I am recommending but I am going to approve this since it is just about organization and not about the code itself :) Great job on this, it is a big branch/task!

dvdherron commented:
> 
> 
> @dvdherron One more tiny change I am recommending but I am going to approve this since it is just about organization and not about the code itself :) Great job on this, it is a big branch/task!

Thanks @stacyk! Made that last organization change so I think this is ready to merge @jgerigmeyer.
